tests/adapters/test_epics_adapter.py

```diff
@@ -0,0 +1,130 @@
+import sys
+from dataclasses import is_dataclass
+
+import pytest
+from mock import MagicMock, Mock, create_autospec
+from mock.mock import call, mock_open, patch
+
+from tickit.core.adapter import ConfigurableAdapter, Interpreter
+from tickit.core.device import Device
+
+# % % % % % % % % % % WARNING % % % % % % % % % %
+# sys.modules["softioc"] = Mock() must be located
+# above the import of tickit.adapters.epicsadapter
+sys.modules["softioc"] = Mock()
+from tickit.adapters.epicsadapter import (  # noqa; isort:skip
+    EpicsAdapter,
+    InputRecord,
+    softioc,
+)
+
+
+@pytest.fixture
+def MockDevice() -> Mock:
+    return create_autospec(Device, instance=False)
+
+
+@pytest.fixture
+def MockInterpreter() -> Mock:
+    return MagicMock(Interpreter, instance=True)
+
+
+@pytest.fixture
+def epics_adapter() -> EpicsAdapter:
+    return EpicsAdapter("db_file", "ioc_name")
+
+
+@pytest.fixture
+def input_record() -> InputRecord:
+    def setter():
+        return None
+
+    def getter():
+        return False
+
+    return InputRecord("input", Mock(setter), Mock(getter))
+
+
+def test_epics_adapter_is_configurable_adapter():
+    assert issubclass(EpicsAdapter, ConfigurableAdapter)
+
+
+def test_epics_adapter_configures_dataclass():
+    assert is_dataclass(EpicsAdapter.EpicsAdapterConfig)
+
+
+def test_epics_adapter_config_configures_epics_adapter():
+    assert EpicsAdapter.EpicsAdapterConfig.configures() is EpicsAdapter
+
+
+def test_epics_adapter_constuctor(epics_adapter: EpicsAdapter):
+    pass
```

Comment:
This seems superfluous

Reply:
It is redundant, but it's often the first thing I test. Besides the constructor is a method that can be broken if it is modified. Although a broken constructor would fail other tests, **only** a broken constructor can fail this test.

Reply:
Ah, so you're just delegating the test to the fixture? In these cases I've chosen to duplicate code across both the test and the fixture with the thought that any changes have to be substantially more deliberate. Not sure what's considered good practice though

Reply:
Potentially having a proper test for constructors is better. I suppose you could parametrize them for a more thorough test. But most of the ones in this PR just assign args to attributes so there's not much point.